operator/pkg/controllers/utils.go

```diff
@@ -586,3 +586,86 @@ func applyRateLimitToContainerArg(container map[string]interface{}, rateLimit *c
 	}
 	return nil
 }
+
+func ApplyContainerPprofConfig(m *manifest.Objects, targetControllerName string, pprofConfig *customizev1alpha1.PprofConfig) error {
+	if pprofConfig == nil {
+		return nil
+	}
+
+	var (
+		targetContainerName string
+		targetControllerGVK schema.GroupVersionKind
+	)
+	switch targetControllerName {
+	case "cnrm-controller-manager":
+		targetContainerName = "manager"
+		targetControllerGVK = schema.GroupVersionKind{
+			Group:   appsv1.SchemeGroupVersion.Group,
+			Version: appsv1.SchemeGroupVersion.Version,
+			Kind:    "StatefulSet",
+		}
+	default:
+		return fmt.Errorf("pprof config customization for %s is not supported. "+
+			"Supported controllers: %s",
+			targetControllerName, strings.Join(customizev1alpha1.SupportedPprofControllers, ", "))
+	}
+
+	for _, item := range m.Items {
+		if item.GroupVersionKind() != targetControllerGVK {
+			continue
+		}
+		if !strings.HasPrefix(item.GetName(), targetControllerName) {
+			continue
+		}
+		if err := item.MutateContainers(customizePprofConfigFn(targetContainerName, pprofConfig)); err != nil {
+			return err
+		}
+		break // we already found the matching controller, no need to keep looking.
+	}
+	return nil
+}
+
+func customizePprofConfigFn(target string, pprofConfig *customizev1alpha1.PprofConfig) func(container map[string]interface{}) error {
+	return func(container map[string]interface{}) error {
+		name, _, err := unstructured.NestedString(container, "name")
+		if err != nil {
+			return fmt.Errorf("error reading container name: %w", err)
+		}
+		if name != target {
+			return nil
+		}
+		return applyPprofConfigToContainerArg(container, pprofConfig)
+	}
+}
+
+func applyPprofConfigToContainerArg(container map[string]interface{}, pprofConfig *customizev1alpha1.PprofConfig) error {
+	if pprofConfig == nil {
+		return nil
+	}
+	origArgs, found, err := unstructured.NestedStringSlice(container, "args")
+	if err != nil {
+		return fmt.Errorf("error getting args in container: %w", err)
+	}
+	wantArgs := []string{}
```

Comment:
LGTM.  Somewhere we have a similar function to replace flags, but I don't know which codebase that is in!